### PR TITLE
feat: Add resource tags that indicate engine used by context

### PR DESCRIPTION
**Pull request by mattcarp12**
Issue #, if available:

**Description of Changes**

[//]: #  (A description of the change that you made and the new user experience that it creates)

**Description of how you validated changes**

[//]: #  (A description of you validated your changes and any relevant logs that show your change works)

**Checklist**

- [x] If this change would make any existing documentation invalid, I have included those updates within this PR
- [x] I have added unit tests that prove my fix is effective or that my feature works
- [x] I have linted my code before raising the PR
- [x] Title of this Pull Request follows Conventional Commits standard: https://www.conventionalcommits.org/en/v1.0.0/

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/cdk/lib/env/context-app-parameters.ts [hunk 1]
```diff
@@ -212,4 +218,28 @@ export class ContextAppParameters {
     }
     return defFilesystem;
   }
+
+  public getEngineType(): string {
+    let engineType: string;
+    switch (this.engineName) {
```
Comment (vvalleru) [inline packages/cdk/lib/env/context-app-parameters.ts:224]: Should you ignore case before checking equivalency?

File: packages/cdk/lib/env/context-app-parameters.ts [hunk 2]
```diff
@@ -212,4 +218,28 @@ export class ContextAppParameters {
...
+        engineType = "wdl";
+        break;
+      case "snakemake":
+        engineType = "snakemake";
+        break;
+      case "toil":
+        engineType = "cwl";
+        break;
+      default:
+        throw Error(`Engine '${this.engineName}' is not supported`);
```
Comment (vvalleru) [inline packages/cdk/lib/env/context-app-parameters.ts:241]: Does it need to be blocking? Just a log with a warn or a “unknown” value should be enough. Thoughts?

File: packages/cdk/lib/env/context-app-parameters.ts (hunk 1 above)
Comment (mattcarp12) [reply packages/cdk/lib/env/context-app-parameters.ts:224]: Here I was following the pattern of `getDefaultFilesystem()` above, which does not ignore case. Do you suggest I change both to ignore case?

File: packages/cdk/lib/env/context-app-parameters.ts (hunk 2 above)
Comment (mattcarp12) [reply packages/cdk/lib/env/context-app-parameters.ts:241]: Here just following the pattern of `getDefaultFilesystem()` above, which throws the error. For me, I think it's fine to leave as is.

Comment (vvalleru) [reply packages/cdk/lib/env/context-app-parameters.ts:241]: Tagging the resources is for our analytics and not a "customer use case". So, I we should not block a customer use case (stack deployment in this case) when we fail to associate analytical data.

File: packages/cdk/lib/env/context-app-parameters.ts (hunk 1 above)
Comment (vvalleru) [reply packages/cdk/lib/env/context-app-parameters.ts:224]: Unlike `filesystem`, the engine name is determined from the customer's `agc-project.yml` file. Its good to have some fool proof. If customer uses case in their project files, we should still be able to determine and tag the resources with the right engine metadata.

**Review by vvalleru — CHANGES_REQUESTED** [review]

File: packages/cdk/lib/env/context-app-parameters.ts (hunk 1 above)
Comment (mattcarp12) [reply packages/cdk/lib/env/context-app-parameters.ts]: Good point, will update.

File: packages/cdk/lib/env/context-app-parameters.ts (hunk 2 above)
Comment (mattcarp12) [reply packages/cdk/lib/env/context-app-parameters.ts]: Yes I agree.

**Review by vvalleru — APPROVED** [review]

**Pull request merged**